### PR TITLE
Remove workaround for issue #37674

### PR DESCRIPTION
**Pull request by frankbu**
Please provide a description for what this PR is for.

depends on https://github.com/istio/istio/pull/40304

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by ericvn** [comment]
@frankbu Are you planning to cherry-pick https://github.com/istio/istio/pull/40304 to release 1.15? Else we won't merge this until after 1.15 is released.

**Comment by frankbu** [comment]
@ericvn good point. It would be nice to get this into 1.15, last of the enhancements and cleanups for external control plane. I'll cherrypick the fix.

**Review by ericvn — APPROVED** [review]

**Pull request merged**
